### PR TITLE
feat: deselect option from dropdown in single select

### LINKED ISSUE #2053: Allow deselect of option when [multiple]="false"
**Is your feature request related to a problem? Please describe.**
It seems to me that the ng-select for `[multiple]="true"` should behave the same as the ng-select for `[multiple]="false"`.
However, it doesn't seem possible to deselect an option from the dropdown when only one option can be selected.

**Describe the solution you'd like**
What I would like to have is an option, such as `@Input() clickOptionToDeselect: boolean` for the ng-select, that can toggle this behaviour. This could apply to ng-selects that allow multiple options as well as ng-selects with only a single selected option allowed.

When `false`, clicking on an `<ng-option>` would not deselect it. It only selects the option if it isn't currently selected.
When `true`, clicking on an `<ng-option>` would toggle the selection, regardless of the `multiple` property.

When the `multiple` property is `true`, the value of the proposed `clickOptionToDeselect` would be `true`; when `multiple` is set to `false`, the default value could be `false`, which would then mimic the current behaviour and ensure that current usage doesn't have to change if it is not needed.

**Describe alternatives you've considered**
I've considered rewriting my implementation using the `ng-option-tmp` template, however this makes for a very dirty fix for this issue, while it would be much simpler to implement it in the library.

This implementation would consist of adding a click listener to a `div` inside the template, which would call the `handleClearClick()` method of the ng-select. However, this would not actually change the value of the ng-select for some reason. Only after adding a `close()` after the `handleClearClick()` would the value of the ng-select change.

Therefore this feature in the library would make more sense.

**Additional context**
Some additional notes: in my project, the clear-all button is removed because of the styling, and I'm trying to fix this without putting that button back in.

### PR DESCRIPTION
**Pull request by Tlepel**
This feature enables deselecting an option from the dropdown when `multiple` is `false`. The user can
click on a selected option, which will deselect that option instead of closing the dropdown.

Fixes #2053

### CODE REVIEW

**Comment by Tlepel** [comment]
The linked issue is closed, however I still think a fix would be nice. I've made a workaround half a year ago, but just now I came across the same problem. Re-applying the workaround doesn't feel right, as it adds so much clutter to the code base.

**Comment by Tlepel** [comment]
Also: this is my first PR to `ng-select`, so any feedback is welcome! I'm happy to adapt and improve wherever I can!

**Comment by anurag-roy** [comment]
@varnastadeus This would be a great feature to have. Can this please be reviewed and merged?

**Review by anurag-roy — APPROVED** [review]

**Comment by anurag-roy** [comment]
@pavankjadda CI is green. Can this be merged?

**Comment by Tlepel** [comment]
@pavankjadda the auto-merge doesn't seem to work properly, since this PR needs an approval before it can be merged 😅

File: src/ng-select/lib/ng-select.component.ts
Comment (pavankjadda) [inline src/ng-select/lib/ng-select.component.ts:455]: Without going through deep into this, why is this removed and next line added?

**Review by pavankjadda — CHANGES_REQUESTED** [review]

File: src/ng-select/lib/ng-select.component.ts
Comment (Tlepel) [reply src/ng-select/lib/ng-select.component.ts:455]: I swapped out `this.multiple` for `this.deselectOnClick` because it makes the code more readable.

If `deselectOnClick` wasn't provided to the component, it takes the value of `this.multiple`, and in that case the behaviour remains the same as before. However, if the component received a value for `deselectOnClick`, that value will be used instead of course.

**Comment by pavankjadda** [comment]
@Tlepel LGTM. Can you add test cases for both scenarios so that I can merge this?

**Comment by Tlepel** [comment]
Sure! I added several tests, one for each combinations of `multiple` and `deselectOnClick`

**Review by pavankjadda — APPROVED** [review]

**Pull request merged**
